File: benchmark/c/cpw/count_digits_factorial_set_1/count_digits_factorial_set_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int count_digits_factorial_set_1 ( int n ) ;
int count_digits_factorial_set_1 ( int n ) {
  if ( n < 0 ) return 0;
  if ( n <= 1 ) return 1;
  double digits = 0;
  for ( int i = 2;
  i <= n;
  i ++ ) digits += log10 ( i );
  return floor ( digits ) + 1;
}

File: translation_result/else_bms/success/c/cpw/count_digits_factorial_set_1/gpt4/Restart/count_digits_factorial_set_1.rs

use std::f64;

fn count_digits_factorial_set_1(n: i32) -> i32 {
    if n < 0 {
        return 0;
    }
    if n <= 1 {
        return 1;
    }
    let mut digits = 0.0;
    for i in 2..=n {
        digits += (i as f64).log10();
    }
    (digits.floor() as i32) + 1
}
